src/libs/actions/Report.ts

```diff
@@ -2515,20 +2578,98 @@ function inviteToRoom(reportID: string, inviteeEmailsToAccountIDs: Record<string
             value: {
                 participantAccountIDs: report.participantAccountIDs,
                 visibleChatMemberAccountIDs: report.visibleChatMemberAccountIDs,
+                participants: inviteeAccountIDs.reduce((revertedParticipants: Record<number, null>, accountID) => {
+                    // eslint-disable-next-line no-param-reassign
+                    revertedParticipants[accountID] = null;
+                    return revertedParticipants;
+                }, {}),
                 pendingChatMembers: report?.pendingChatMembers ?? null,
             },
         },
         ...newPersonalDetailsOnyxData.finallyData,
     ];
 
+    if (ReportUtils.isGroupChat(report)) {
+        const parameters: InviteToGroupChatParams = {
+            reportID,
+            inviteeEmails,
+            accountIDList: newAccountIDs.join(),
+        };
+
+        API.write(WRITE_COMMANDS.INVITE_TO_GROUP_CHAT, parameters, {optimisticData, successData, failureData});
+        return;
+    }
+
     const parameters: InviteToRoomParams = {
         reportID,
         inviteeEmails,
     };
 
+    // eslint-disable-next-line rulesdir/no-multiple-api-calls
     API.write(WRITE_COMMANDS.INVITE_TO_ROOM, parameters, {optimisticData, successData, failureData});
 }
 
+function updateGroupChatMemberRoles(reportID: string, accountIDList: number[], role: ValueOf<typeof CONST.REPORT.ROLE>) {
+    const participants: Participants = {};
+    const memberRoles: Record<number, string> = {};
+    accountIDList.forEach((accountID) => {
+        memberRoles[accountID] = role;
+        participants[accountID] = {role};
+    });
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: {
+                participants,
+            },
+        },
+    ];
+
+    API.write(WRITE_COMMANDS.UPDATE_GROUP_CHAT_MEMBER_ROLES, {reportID, memberRoles: JSON.stringify(memberRoles)}, {optimisticData});
```

Comment:
Similar to other commands there isn't any expected error handling to deal with. We could minimally revert back to the previous role. But not too passionate about having any specific pending offline behavior for something small like changing the Group Chat role. Seems like we could achieve it with the `pendingChatMembers` and also add something like `pendingAction: changeRole`, but I had not planned to do this.